homeassistant/helpers/service.py

```diff
@@ -934,28 +939,27 @@ async def _handle_entity_call(
 
     task: asyncio.Future[ServiceResponse] | None
     if isinstance(func, str):
-        task = hass.async_run_job(
-            partial(getattr(entity, func), **data)  # type: ignore[arg-type]
+        task = hass.async_run_hass_job(
+            HassJob(partial(getattr(entity, func), **data))  # type: ignore[arg-type]
         )
     else:
-        task = hass.async_run_job(func, entity, data)
+        task = hass.async_run_hass_job(HassJob(func), entity, data)
```

Comment:
Ideally we would accept `HassJob` directly so we don't have to determine how its run but I wanted to fix the typing first before I thought about how to do that